airflow/jobs/scheduler_job_runner.py

```diff
@@ -722,6 +722,9 @@ def _process_executor_events(self, session: Session) -> int:
                 self.log.info("Setting external_id for %s to %s", ti, info)
                 continue
 
+            if ti.state in State.finished:
+                ti.dag_run.activate_scheduling()
```

Comment:
Should we handle terminal tasks any differently?

Reply:
Yes. Here is when we enable scheduling for the dagrun. Likely, the task that has completed deactivated the dagrun, so we need to re-enable scheduling